### PR TITLE
feat: OB-39910 add multiple metric streams to work around 1000 metric…

### PR DESCRIPTION
**Pull request by obs-gh-virjramakrishnan**
… limit

### CODE REVIEW

File: apps/metricstream/template.yaml [hunk 1]
```diff
@@ -224,6 +226,7 @@ Resources:
                 Resource: !GetAtt 'DeliveryStream.Arn'
   MetricStream:
     Type: AWS::CloudWatch::MetricStream
+    Condition: NotDeployLambda
```
Comment (obs-gh-alexlew) [inline apps/metricstream/template.yaml:229]: do we need to add a whole variable for this? can't we just do `!DeployLambda`

File: pkg/handler/metricsconfigurator/handler.go [hunk 2]
```diff
@@ -261,6 +270,26 @@ func (h Handler) getDatasource(token *string, observeDomainName string, client *
 	return bodyBytes, nil
 }
 
+func (h Handler) makeMetricGroups(MetricsFilters []types.MetricStreamFilter) [][]types.MetricStreamFilter {
+	// create appropriate number of metric streams for metrics, assign metrics to streams
+
+	currentNameCount := 0
+	filterGroups := make([][]types.MetricStreamFilter, 0)
+	currentFilterGroup := make([]types.MetricStreamFilter, 0)
+	for _, filter := range MetricsFilters {
```
Comment (obs-gh-alexlew) [inline pkg/handler/metricsconfigurator/handler.go:279]: feels like we could do a flatten and then a chunk here no? https://pkg.go.dev/slices@master#Chunk

File: apps/metricstream/template.yaml (hunk 1 above)
Comment (obs-gh-virjramakrishnan) [reply apps/metricstream/template.yaml:229]: I tried `Condition !Not DeployLambda` and the sam app threw an error - it seems it wants the condition to be a string that refers to an existing condition.

File: pkg/handler/metricsconfigurator/handler.go (hunk 2 above)
Comment (obs-gh-virjramakrishnan) [reply pkg/handler/metricsconfigurator/handler.go:279]: ah, I didn't know this existed, but it's much cleaner, I'll refactor accordingly!

Comment (obs-gh-virjramakrishnan) [reply pkg/handler/metricsconfigurator/handler.go:279]: Ah wait, I think that Chunk would not work here - two reasons:
1. We would end up with metrics from the same namespace in different streams. We'd need to make a `types.MetricStreamFilter` for the same namespace in two different streams.
2. AWS counts the namespace itself towards the 1000 (see [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricStreamFilter.html)) so there would be some pretty gnarly cases (adding the namespaces in takes us over the boundary).

**Review by obs-gh-alexlew — APPROVED** [review]

File: pkg/handler/metricsconfigurator/handler.go (hunk 2 above)
Comment (obs-gh-alexlew) [reply pkg/handler/metricsconfigurator/handler.go:279]: yeah so then this basically boils down to binpacking, theoretically its possible to be more efficient but we don't really care about how many metric streams it takes us so this simple approach is probably best

**Pull request merged**